### PR TITLE
Menu set?

### PR DESCRIPTION
**Pull request by idawud**
are we set to go? what should be added or removed

### CODE REVIEW

File: src/main/java/io/turntabl/TCMS.java [hunk 1]
```diff
@@ -1,21 +1,17 @@
...
 public class TCMS {
     public static void main(String[] args) throws IOException {
-                // ClientInformationPersistence cip = new ClientInformationPersistence();
-        // Files.readAllLines(Paths.get("./resources/clientsInformation.txt")).forEach(System.out::println);
-        // cip.retrieveAll().forEach(clientData -> System.out.println(clientData.toCSV()));
-        // System.out.println(cip.store("last killer", "nankrom", "558-8958-65", "mail@mail.er") );
-        // cip.search("babon").forEach(clientData -> System.out.print(clientData.toCSV()));
-        // cip.delete(3);
-        // cip.retrieveAll().forEach(clientData -> System.out.println(clientData.toCSV()));
+        ClientInformationPersistence cip = new ClientInformationPersistence();
```
Comment (patriciakasapa) [inline src/main/java/io/turntabl/TCMS.java:11]: How about you change the cip to something meaningful for clarity sake?

File: src/main/java/io/turntabl/persistance/ClientInformationPersistence.java [hunk 2]
```diff
@@ -80,7 +80,7 @@ private String clientDataToCsvString(ClientData clientData){
     public List<ClientData> search(String name) throws IOException {
         List<String> allData = readFile();
         return allData.stream()
-                .filter(s -> (s.split(",")[1]).equals(name))
+                .filter(s -> (s.split(",")[1]).toLowerCase().equals(name.toLowerCase()))
```
Comment (idawud) [inline src/main/java/io/turntabl/persistance/ClientInformationPersistence.java:83]: i should trim the string too for comparison

File: src/main/java/io/turntabl/dataaccess/DataAccess.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
+package io.turntabl.dataaccess;
+
```
Comment (patriciakasapa) [inline src/main/java/io/turntabl/dataaccess/DataAccess.java:2]: Great job... Neat presentation!!!

File: src/main/java/io/turntabl/TCMS.java [hunk 4]
```diff
@@ -1,21 +1,17 @@
...
 
 public class TCMS {
     public static void main(String[] args) throws IOException {
-                // ClientInformationPersistence cip = new ClientInformationPersistence();
-        // Files.readAllLines(Paths.get("./resources/clientsInformation.txt")).forEach(System.out::println);
-        // cip.retrieveAll().forEach(clientData -> System.out.println(clientData.toCSV()));
-        // System.out.println(cip.store("last killer", "nankrom", "558-8958-65", "mail@mail.er") );
-        // cip.search("babon").forEach(clientData -> System.out.print(clientData.toCSV()));
-        // cip.delete(3);
-        // cip.retrieveAll().forEach(clientData -> System.out.println(clientData.toCSV()));
```
Comment (pkErbynn) [inline src/main/java/io/turntabl/TCMS.java:15]: yo bubbler, why did you get rid of these ?

File: src/main/java/io/turntabl/dataentry/DataEntry.java [hunk 5]
```diff
@@ -29,7 +29,7 @@ public  static boolean toSave() throws IOException {
         return Arrays.asList(name, address, telephoneNumber, email);
     }
 
-    private static String getStringInput(String s) {
+    public static String getStringInput(String s) {
```
Comment (patriciakasapa) [inline src/main/java/io/turntabl/dataentry/DataEntry.java:32]: Great!

File: src/main/java/io/turntabl/TCMS.java (hunk 1 above)
Comment (pkErbynn) [reply src/main/java/io/turntabl/TCMS.java:11]: oh nooooo! 

> yeah, are right

File: src/main/java/io/turntabl/TCMS.java (hunk 4 above)
Comment (patriciakasapa) [reply src/main/java/io/turntabl/TCMS.java:15]: Lol... We had to merge them in another file... They still live

File: src/main/java/io/turntabl/menu/AnsiConsole.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package io.turntabl.menu;
+
+public class AnsiConsole {
+    static final String RESET = "\033[0m";                  // Text Reset
+
```
Comment (patriciakasapa) [inline src/main/java/io/turntabl/menu/AnsiConsole.java:5]: Oh yeah... I really like the color combo... They are on point

File: src/main/java/io/turntabl/dataaccess/DataAccess.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+import java.io.IOException;
+import java.util.List;
+
+public class DataAccess {
+
+    public static void showAllClientsRecords() throws IOException {
+        ClientInformationPersistence cip = new ClientInformationPersistence();
+        List<ClientData> records = cip.retrieveAll();
+        records.forEach(Printer::printClientCard);
```
Comment (pkErbynn) [inline src/main/java/io/turntabl/dataaccess/DataAccess.java]: `we could have used lambda expression here`

File: src/main/java/io/turntabl/dataaccess/DataAccess.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    public static void deleteClientRecord() throws IOException {
+        String name = DataEntry.getStringInput("Enter Client's name: ");
+        ClientInformationPersistence cip = new ClientInformationPersistence();
+        cip.search(name).forEach(Printer::printClientCard);
+
+        int id = 9999;
+        if (cip.delete(id)){
+            System.out.println("Client Record Deleted Successfully!");
```
Comment (pkErbynn) [inline src/main/java/io/turntabl/dataaccess/DataAccess.java]: we can and a delete prompt here

File: src/main/java/io/turntabl/dataentry/DataEntry.java [hunk 9]
```diff
@@ -29,7 +29,7 @@ public  static boolean toSave() throws IOException {
...
         System.out.println(s);
         return ( INPUT.nextLine() );
```
Comment (pkErbynn) [inline src/main/java/io/turntabl/dataentry/DataEntry.java:34]: what is the meaning of the `INPUT`.......an enum

File: src/main/java/io/turntabl/menu/MainMenu.java [hunk 10]
```diff
@@ -12,13 +13,13 @@ public static void operation(int option) throws IOException {
...
+                DataAccess.showAllClientsRecords();
                 break;
             case 3:
-                System.out.println("Searching for client");
+                DataAccess.showSearchedClientsRecords();
                 break;
             case 4:
-                System.out.println("Deleting a client");
+                DataAccess.deleteClientRecord();
                 break;
```
Comment (pkErbynn) [inline src/main/java/io/turntabl/menu/MainMenu.java:23]: nice one there !

File: src/main/java/io/turntabl/menu/MainMenu.java [hunk 11]
```diff
@@ -12,13 +13,13 @@ public static void operation(int option) throws IOException {
...
             case 5:
                 System.exit(0);
```
Comment (pkErbynn) [inline src/main/java/io/turntabl/menu/MainMenu.java:25]: maybe we could use a **default** here

**Review by patriciakasapa — COMMENTED** [review]
Great job done, guys!
We are making significant progress...
Pizza on me when i build my first house!

File: src/main/java/io/turntabl/persistance/ClientInformationPersistence.java [hunk 12]
```diff
@@ -80,7 +80,7 @@ private String clientDataToCsvString(ClientData clientData){
...
                 .map(this::stringToClientData)
                 .collect(Collectors.toList());
     }
```
Comment (pkErbynn) [inline src/main/java/io/turntabl/persistance/ClientInformationPersistence.java:86]: looks good to me!

File: src/main/java/io/turntabl/dataentry/DataEntry.java (hunk 9 above)
Comment (pkErbynn) [reply src/main/java/io/turntabl/dataentry/DataEntry.java:34]: * enum ?

File: src/main/java/io/turntabl/dataaccess/DataAccess.java (hunk 8 above)
Comment (patriciakasapa) [reply src/main/java/io/turntabl/dataaccess/DataAccess.java]: My guy...
Wey kind English this?

File: src/main/java/io/turntabl/dataaccess/DataAccess.java (hunk 7 above)
Comment (patriciakasapa) [reply src/main/java/io/turntabl/dataaccess/DataAccess.java]: Well... you can use foreach on a list without necessarily using a stream. It works perfectly.

File: src/main/java/io/turntabl/menu/Printer.java [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+                + AnsiConsole.WHITE_BOLD + "** Address: \t\t\t" + AnsiConsole.RESET + AnsiConsole.CYAN + record.get(1)+ AnsiConsole.RESET );
+        System.out.println("***************************************************************************");
+    }
+
+    public static void printClientCardWithId(ClientData clientData){
+        System.out.println(clientData.toList().get(0));
+    }
+
+    public static void recordNotFound() {
+        System.out.println("Error 404: ClientNotFound");
```
Comment (pkErbynn) [inline src/main/java/io/turntabl/menu/Printer.java]: make `Error 404: ` red with ansi code 

` public static final String RED_BOLD = "\033[1;31m" `

**Review by pkErbynn — APPROVED** [review]
in line 33:
`System.out.println("Client Added Successfully!!!");`

make only the _Successfully_ green

File: src/main/java/io/turntabl/menu/Printer.java (hunk 13 above)
Comment (idawud) [reply src/main/java/io/turntabl/menu/Printer.java]: thanks! done

**Pull request merged**